### PR TITLE
kubetail: add a max lifetime for tailing logs from a container

### LINKED ISSUE #3093: Flow: no log lines when using loki.source.kubernetes or loki.source.podlogs
In my test Kubernetes environment, I've noticed that log lines disappear after a few hours when using `loki.source.kubernetes`. 

I've tracked this down to a network connectivity issue with the connection to the Kubernetes API, where the component is waiting forever for a new log line to appear. 

Since `loki.source.kubernetes` and `loki.source.podlogs` share code for retrieving log lines, this bug affects both.

### PR DESCRIPTION
**Pull request by rfratto**
This adds a maximum lifetime (1 hour) for each tailing connection to a container. This fixes an issue where the connection between the agent, the Kubernetes API server, and the Kubelet can get disrupted without closing the TCP connection, leading to no logs being read.

This bug fix impacts `loki.source.kubernetes` and `loki.source.podlogs`, which both use the kubetail package. 

Closes #3093.

### CODE REVIEW

File: component/loki/source/kubernetes/kubetail/tailer.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func newLabelSet(l labels.Labels) model.LabelSet {
 }
 
 var retailBackoff = backoff.Config{
-	MinBackoff: time.Second,
+	MinBackoff: 0,
```
Comment (rfratto) [inline component/loki/source/kubernetes/kubetail/tailer.go:82]: I set this back to zero so the very first retry is instantaneous, which is useful to make sure that the maximum lifetime of the tailer doesn't have to wait a second before reestablishing the connection.

**Comment by rfratto** [comment]
I deployed this locally and found that there are still some issues; mainly it seems like containers were falsely considered terminated after the lifetime expired. I'm still looking into this.

**Review by erikbaranowski — APPROVED** [review]
LGTM, nice catch on the UID!

**Comment by rfratto** [comment]
I'm going to merge this and continue to run the fix in my Kubernetes cluster. If it still looks good after a couple of days, it would probably make sense to put this fix in a patch release (given the severity of the bug).

**Pull request merged**